megatron/model/init_functions.py

```diff
@@ -2,28 +2,40 @@
 
 import torch
 
+try:
```

Comment:
This import will still fail. Mup cannot be a top-level import (we only want an exception if (mup is not installed) and (we try to use mup)

Reply:
What precisely is wrong with making MuP a requirement? Like, under what conditions is expecting someone to have access to it unreasonable? Or does it have negative externalities I am unaware of?

Reply:
My argument is to reduce requirements bloat. I don't expect many users to require mup, so I don't want it to be required. I consider mup to be similar to our bnb support: "A specialized optimization that's mostly for advanced internal users"